src/kvstore/LogEncoder.cpp

```diff
@@ -167,6 +167,28 @@ HostAddr decodeLearner(const std::string& encoded) {
     return addr;
 }
 
+std::string encodeTransLeader(const HostAddr& learner) {
+    std::string encoded;
+    encoded.reserve(kHeadLen + sizeof(HostAddr));
+    // Timestamp (8 bytes)
+    int64_t ts = time::WallClock::fastNowInMilliSec();
+    encoded.append(reinterpret_cast<char*>(&ts), sizeof(int64_t));
+    // Log type
+    auto type = LogType::OP_TRANS_LEADER;
+    encoded.append(reinterpret_cast<char*>(&type), 1);
```

Comment:
It seems you forget to encode value length here, it will crash when decode when check length, so as encodeAddLearner.